src/app/settings/sponsoring-org-row.component.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    Component,
+    EventEmitter,
+    Input,
+    Output,
+} from '@angular/core';
+import { ToasterService } from 'angular2-toaster';
+import { ApiService } from 'jslib-common/abstractions/api.service';
+import { I18nService } from 'jslib-common/abstractions/i18n.service';
+import { LogService } from 'jslib-common/abstractions/log.service';
+import { PlatformUtilsService } from 'jslib-common/abstractions/platformUtils.service';
+
+import { Organization } from 'jslib-common/models/domain/organization';
+
+@Component({
+    selector: '[sponsoring-org-row]',
+    templateUrl: 'sponsoring-org-row.component.html',
+})
+export class SponsoringOrgRowComponent {
+    @Input() sponsoringOrg: Organization = null;
+
+    @Output() sponsorshipRemoved = new EventEmitter();
+
+    loading = true;
+
+    revokeSponsorshipPromise: Promise<any>;
+    resendEmailPromise: Promise<any>;
+
+    constructor(private toasterService: ToasterService, private apiService: ApiService,
+        private i18nService: I18nService, private logService: LogService,
+        private platformUtilsService: PlatformUtilsService) { }
+
+    ngOnInit() {
+        if (this.sponsoringOrg === null) {
+            return;
+        }
+
+        this.loading = false;
+    }
+
+    async revokeSponsorship() {
+        try {
+            this.revokeSponsorshipPromise = this.doRevokeSponsorship();
+            await this.revokeSponsorshipPromise;
+        } catch (e) {
+            this.logService.error(e);
+        }
+
+        this.revokeSponsorshipPromise = null;
+    }
+
+    async resendEmail() {
+        this.resendEmailPromise = this.apiService.postResendSponsorshipOffer(this.sponsoringOrg.id);
+        await this.resendEmailPromise;
+        this.toasterService.popAsync('success', null, this.i18nService.t('emailSent'));
+        this.resendEmailPromise = null;
+    }
+
+    private async doRevokeSponsorship() {
+        const isConfirmed = await this.platformUtilsService.showDialog(
+            this.i18nService.t('revokeSponsorshipConfirmation'), 
+            `${this.i18nService.t('remove')} ${this.sponsoringOrg.familySponsorshipFriendlyName}?`,
+            this.i18nService.t('remove'), this.i18nService.t('cancel'), 'warning');
+
+        if (!isConfirmed) {
+            return;
+        }
+
+        await this.apiService.deleteRevokeSponsorship(this.sponsoringOrg.id);
+        this.sponsorshipRemoved.emit();
```

Comment:
Personal preference, but I prefer to have this last in the method. Helps a bit with the mental model that we're exiting this function and returning to the event listener. Vs calling event listener and then depending on js event loop we're returning here before going to the listener.

Reply:
I think I followed but please check again to make sure I did what you preferred.

Reply:
I only wanted to move the `this.sponsorshipRemoved.emit();` below `toast.popAsync` . Didn't even notice the function wasn't in the top of the call trace. It's not a hard rule either.